contents/docs/advanced/proxy/cloudflare.md

```diff
@@ -4,7 +4,78 @@ sidebar: Docs
 
-> Cloudflare does require your domain to be hosted with them, and using them does more than just proxying requests, such as blocking traffic from bots.
-> Additionally, you must be on the Enterprise CloudFlare plan to customize the Host header.
+There are two ways of going about this. The first one – involving Cloudflare Workers – is a bit more setup, but can be used on all Cloudflare plans. The second one – involving Page Rules – is simpler, but requires the Cloudflare Enterprise plan. Choose accordingly.
+
+## A. Proxy using Cloudflare Workers
+
+Workers are really powerful and allow up to 100,000 requests per day on the free plan ([see Cloudflare pricing](https://developers.cloudflare.com/workers/platform/pricing/)). Follow these steps to set up a reverse proxy worker:
+
+### 1. Create a worker
+
+From the root of the Cloudflare dashboard, go to "Workers & Pages" > "Overview" > "Create application" > "Create Worker". At this point, you can either keep the random worker name or choose your own. Click "Deploy" once done.
+
+### 2. Configure the worker to act as a proxy
+
+Click "Edit code" once the new worker has been saved following "Deploy". (And if you're already on the worker page, click "Quick edit".) You should now be seeing a code editor for the worker. Just replace all the existing content with this proxying code:
+
+```JavaScript
+const API_HOST = "app.posthog.com" // Change to "eu.posthog.com" for the EU region
+
+async function handleRequest(event) {
+    const pathname = new URL(event.request.url).pathname
+    if (pathname.startsWith("/static/")) {
+        return retrieveStatic(event, pathname)
+    } else {
+        return forwardRequest(event, pathname)
+    }
+}
+
+async function retrieveStatic(event, pathname) {
+    let response = await caches.default.match(event.request)
+    if (!response) {
+        response = await fetch(`https://${API_HOST}${pathname}`)
+        event.waitUntil(caches.default.put(event.request, response.clone()))
+    }
+    return response
+}
+
+async function forwardRequest(event, pathname) {
+    const request = new Request(event.request)
+    request.headers.delete("cookie")
+    return await fetch(`https://${API_HOST}${pathname}`, request)
+}
+
+addEventListener("fetch", (event) => {
+    event.passThroughOnException()
+    event.respondWith(handleRequest(event))
+})
+```
+
+When done, click "Save and deploy".
+
+### 3. Use a custom domain for the worker
+
+It's optional, but highly recommended, to use your own domain instead of the basic `*.workers.dev` one.
+To do this, go to the worker page (by exiting the code editor, if you're still in it) and there click "View" under "Custom Domains". Just click "Add Custom Domain", type in a subdomain, and save with "Add Custom Domain". The subdomain can be anything, even `pineapple.yourdomain.com` – just remember to avoid terms like "tracking" or "analytics", as they may be blanket-blocked.
```

Comment:
```suggestion
To use your own domain instead of the basic `*.workers.dev` one, go to the worker page (by exiting the code editor, if you're still in it) and there click "View" under "Custom Domains". Click "Add Custom Domain", type in a subdomain, and save with "Add Custom Domain". The subdomain can be anything, even `pineapple.yourdomain.com` – just remember to avoid terms like "tracking" or "analytics", as they may be blanket-blocked.
```